apps/playlist/apis/playlist/v0alpha1/playlist_object_gen.go

```diff
@@ -0,0 +1,265 @@
+//
+// Code generated by grafana-app-sdk. DO NOT EDIT.
+//
+
+package v0alpha1
+
+import (
+	"fmt"
+	"github.com/grafana/grafana-app-sdk/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"time"
+)
+
+// +k8s:openapi-gen=true
+type Playlist struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              PlaylistSpec   `json:"spec"`
+	PlaylistStatus    PlaylistStatus `json:"status"`
+}
+
+func (o *Playlist) GetSpec() any {
+	return o.Spec
+}
+
+func (o *Playlist) SetSpec(spec any) error {
+	cast, ok := spec.(PlaylistSpec)
+	if !ok {
+		return fmt.Errorf("cannot set spec type %#v, not of type Spec", spec)
+	}
+	o.Spec = cast
+	return nil
+}
+
+func (o *Playlist) GetSubresources() map[string]any {
+	return map[string]any{
+		"status": o.PlaylistStatus,
+	}
+}
+
+func (o *Playlist) GetSubresource(name string) (any, bool) {
+	switch name {
+	case "status":
+		return o.PlaylistStatus, true
+	default:
+		return nil, false
+	}
+}
+
+func (o *Playlist) SetSubresource(name string, value any) error {
+	switch name {
+	case "status":
+		cast, ok := value.(PlaylistStatus)
+		if !ok {
+			return fmt.Errorf("cannot set status type %#v, not of type PlaylistStatus", value)
+		}
+		o.PlaylistStatus = cast
+		return nil
+	default:
+		return fmt.Errorf("subresource '%s' does not exist", name)
+	}
+}
+
+func (o *Playlist) GetStaticMetadata() resource.StaticMetadata {
+	return resource.StaticMetadata{
+		Name:      o.ObjectMeta.Name,
+		Namespace: o.ObjectMeta.Namespace,
+		Group:     o.GroupVersionKind().Group,
+		Version:   o.GroupVersionKind().Version,
+		Kind:      o.GroupVersionKind().Kind,
+	}
+}
+
+func (o *Playlist) SetStaticMetadata(metadata resource.StaticMetadata) {
+	o.Name = metadata.Name
+	o.Namespace = metadata.Namespace
+	o.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   metadata.Group,
+		Version: metadata.Version,
+		Kind:    metadata.Kind,
+	})
+}
+
+func (o *Playlist) GetCommonMetadata() resource.CommonMetadata {
```

Comment:
Why is this necessary?  Anything wrong with moving https://github.com/grafana/grafana/blob/main/pkg/services/apiserver/utils/meta.go#L49 to the sdk?

Reply:
@IfSentient i believe this is a question for you

Reply:
It's required for app-sdk because current app-sdk customers depend on get/set `CommonMetadata`. We can deprecate it in favor of something else, but it's currently in-use.